src/lib.rs

```diff
@@ -160,7 +172,7 @@ pub enum Opcode {
 }
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
-enum FieldSpec {
+pub enum FieldSpec {
```

Comment:
Ditto - looks like there isn't a convention here, but I am curious to your use-case.